tidb-best-practices.md

```diff
@@ -0,0 +1,218 @@
+
+```sql
+for i from 0 to 23:
+    while affected_rows > 0:
+        delete * from t where insert_time >= i:00:00 and insert_time < (i+1):00:00 limit 5000;
+        affected_rows = select affected_rows()
+```
+
+This pseudocode means to split huge chunks of data into small ones and then delete, so that the earlier `Delete` statements do not affect the later ones.
+
+### Query
+
+For query requirements and specific statements, refer to [TiDB Specific System Variables](/tidb-specific-system-variables.md).
+
+You can control the concurrency of SQL execution through the `SET` statement and the selection of the `Join` operator through Hint.
+
+In addition, you can also use MySQL's standard index selection, the `Hint` syntax, or control the optimizer to select index through `Use Index/Ignore Index hint`.
+
+If the business scenario has both OLTP and OLAP workloads, you can send the OLTP request and OLAP request to different TiDB servers, diminishing the impact of OLAP on OLTP. It is recommended to use machines with high configurations (e.g. more processor cores, larger memory) for the TiDB server that carries OLAP business.
+
+To completely isolate OLTP and OLAP workloads, it is recommended to run OLAP applications on TiFlash. TiFlash is a columnar storage engine with great performance on OLAP workloads. TiFlash can achieve physical isolation on the storage layer and guarantees consistency read.
+
+### Monitoring and log
+
+The monitoring metrics is the best method to learn the status of the system. It is recommended that you deploy the monitoring system along with your TiDB cluster.
+
+TiDB uses [Grafana + Prometheus](/tidb-monitoring-framework.md) to monitor the system status. The monitoring system is automatically deployed and configured if you deploy TiDB using TiUP.
+
+There are lots of items in the monitoring system, the majority of which are for TiDB developers. You do not have to understand these items without an in-depth knowledge of the source code. Some items that are related to applications or to the state of system key components are selected and put in a separate `overview` panel for users.
+
+In addition to monitoring, you can also view the system logs. The three components of TiDB, tidb-server, tikv-server and pd-server, each has a `--log-file` parameter. If this parameter has been configured when the cluster is started, logs will be stored in the file configured by the parameter and log files are automatically archived on a daily basis. If the `--log-file` parameter has not been configured, log will be output to `stderr`.
```

Comment:
```suggestion
In addition to monitoring, you can also view the system logs. The three components of TiDB, tidb-server, tikv-server, and pd-server, each has a `--log-file` parameter. If this parameter has been configured when the cluster is started, logs are stored in the file configured by the parameter and log files are automatically archived on a daily basis. If the `--log-file` parameter has not been configured, the log is output to `stderr`.
```

Reply:
We usually use the present tense instead of the future tense in technical writing.